TriblerGUI/widgets/downloadspage.py

```diff
@@ -422,6 +448,22 @@ def on_export_download_request_done(self, filename, data):
         else:
             self.window().tray_show_message("Torrent file exported", "Torrent file exported to %s" % dest_path)
 
+    def on_add_to_channel(self):
+        self.request_mgr = TriblerRequestManager()
+        for selected_item in self.selected_items:
+            infohash = selected_item.download_info["infohash"]
+            name = selected_item.download_info["name"]
+            post_data = {"uri": u"magnet:?xt=urn:btih:%s&dn=%s" % (infohash, name)}
```

Comment:
Don't we have a utility method to construct a magnet URL from info hash and name?

Reply:
Good point. Updated